### DISCUSSION TITLE
The draw.io configuration space created by draw.io for confluence 13+

### DISCUSSION THREAD

**Discussion by davidjgraph** [jgraph/drawio#4460] [General]
Upgrading to, or beyond, draw.io 13.0.0 for the first time creates a "draw.io configuration" space.

The draw.io configuration space is needed from release 13.0 onwards because we moved the remaining items stored in the confluence database into this space.

The items now stored in this space are:

- the draw.io JSON configuration

- any custom draw.io libraries

- any custom draw.io templates

- the default templates that come with draw.io (they were previously provided from a different URL).

The reasons for this change are:

- Direct use of the confluence database is discouraged by Atlassian as much as possible, for performance and stability reasons.

- This structure lines up with the storage mechanism draw.io uses in Confluence Cloud. Atlassian’s vision is for all on-premise users to eventually move to cloud.

- The draft storage of in-progress diagrams moves from the database to a page attachment. This makes collaborative editing possible.

**FAQ:**

_Can we delete the space?_

We would strongly advise against this, unless there is a specific reason.

_What are the permissions to the space?_

The config and libraries pages are write restricted to admins. Users can read the config and libraries that they have existing permissions to.

Templates inherit the existing page tree permissions of the previous URL location.

**Comment by deepanudaiyar** [comment]
I am in middle of consolidating mutliple atlassian sites into one centeralized site. As such, few sites have their own drawio config space and these are in conflicts due to space key being the same, while migrating sites to centralized instance. What do you recommend?

**Reply by davidjgraph** [reply]
I would take a copy of each config space, build the central config space manually using the information from the others, but don't actually migrate any of the drawio config spaces using the space migration tool.
